pyccel/codegen/printing/fcode.py

```diff
@@ -1176,14 +1173,22 @@ def _print_AliasAssign(self, expr):
         return self._get_statement(code) + '\n'
 
     def _print_CodeBlock(self, expr):
-        body = []
-        for b in expr.body:
+        body_exprs, new_vars = expand_to_loops(expr.body, language_has_vectors = True)
+        if self._current_function:
+            name = self._current_function
+            func = self.get_function(name)
+            func.local_vars.extend(new_vars)
```

Comment:
@saidctb This block seems to be causing trouble with `__init__` functions. I am getting the error:
```
ERROR at code generation stage
pyccel:
 |fatal [codegen]: generic_methods.py| Undefined function (__init__)

```
Do you have any idea how to fix this?

Reply:
there is something strange happening there, __init__ should not be there, if you can delete that part it will be great;I'm planing on deleting the scope dependency

Reply:
Why should `__init__` not be there?
Is it not a FunctionDef?

Reply:
yes there is no function with name `__init__` , unless you are testing classes

Reply:
The codegen tests test classes so yes I am

Reply:
I have added a `self._current_class` attribute which works similarly to `self._current_function` (See 2621994)
Does this make sense to you?